restricted_site_access.php

```diff
@@ -607,6 +795,24 @@ public static function deactivation() {
 			update_option( 'blog_public', 1 );
 		}
 	}
+
+	public static function is_network( $plugin ) {
+
+		$plugins = get_site_option( 'active_sitewide_plugins');
+
+		if ( is_multisite() && isset( $plugins[ $plugin ] ) ) {
+			return true;
+		}
+
+		return false;
+
+	}
+}
+
+$network_activated = Restricted_Site_Access::is_network( plugin_basename( __FILE__ ) );
+
+if ( $network_activated ) {
+	define( 'RSA_IS_NETWORK', true );
```

Comment:
Just define RSA_IS_NETWORK to false otherwise then you simplify the IF check throughout the code